projects/common/src/main/java/dan200/computercraft/shared/pocket/apis/PocketAPI.java

```diff
@@ -105,19 +109,19 @@ private static void storeItem(Player player, ItemStack stack) {
         }
     }
 
-    private static @Nullable IPocketUpgrade findUpgrade(NonNullList<ItemStack> inv, int start, @Nullable IPocketUpgrade previous) {
+    private static @Nullable ItemStack findUpgrade(NonNullList<ItemStack> inv, int start, @Nullable UpgradeData<IPocketUpgrade> previous) {
```

Comment:
Can we make this return an `UpgradeData<IPocketUpgrade>` instead? This would mean we don't need to do the lookup again in `equipBack`.

Reply:
Sure, done